lib/tests/streamlit/delta_generator_test.py

```diff
@@ -208,6 +211,45 @@ def set_widget_requires_args(self):
         c = self.get_delta_from_queue().new_element.exception
         self.assertEqual(c.type, "TypeError")
 
+    def test_duplicate_widget_id_error(self):
+        """Multiple widgets with the same key should report an error."""
+        widgets = {
+            "button": lambda key=None: st.button("", key=key),
+            "checkbox": lambda key=None: st.checkbox("", key=key),
+            "multiselect": lambda key=None: st.multiselect("", options=[1, 2], key=key),
+            "radio": lambda key=None: st.radio("", options=[1, 2], key=key),
+            "selectbox": lambda key=None: st.selectbox("", options=[1, 2], key=key),
+            "slider": lambda key=None: st.slider("", key=key),
+            "text_area": lambda key=None: st.text_area("", key=key),
+            "text_input": lambda key=None: st.text_input("", key=key),
+            "time_input": lambda key=None: st.time_input("", key=key),
+            "date_input": lambda key=None: st.date_input("", key=key),
```

Comment:
How are we going to remember to add new widgets here?

Reply:
I'm not sure there's a good mechanism for this, other than vigilance? That is, anyone adding a new widget should be following the conventions established by all the other widgets. 

We don't currently annotate any of the widget creation functions (and I believe @tvst is hoping to move away from our more deeply-nested decorator abuse within DeltaGenerator?). But: open to suggestions!

Reply:
Yeah... I see abuse in the decorator pattern. I feel having all this logic in `DeltaGenerator` is a bit overwhelming... Maybe better `class` structure? I am hesitant to propose a "base" class/"derived" class mechanism or an equivalent composition pattern, but it may work here.

Reply:
I'll open a ticket to start thinking about how this might be improved!